parsons/etl_helpers/conversions.py

```diff
@@ -0,0 +1,188 @@
+"""
+This module contains functions that can be passed to the Parsons Table convert_column
+method. They take in an original value and return a new value. They can, of course, also be used
+on their own.
+"""
+
+from dateutil.parser import parse
+import datetime
+import re
+
+
+# Datetime conversion functions
+
+
+def date_to_timestamp(value, tzinfo=datetime.timezone.utc):
+    """Convert any date value into a Unix timestamp.
+
+    `Args:`
+        value: int or str or datetime
+            Value to parse
+        tzinfo: datetime.timezone
+            `Optional`: Timezone for the datetime; defaults to UTC.
+    `Returns:`
+        Unix timestamp (int)
+    """
+
+    parsed_date = parse_date(value)
+
+    if not parsed_date:
+        return None
+
+    if not parsed_date.tzinfo:
+        parsed_date = parsed_date.replace(tzinfo=tzinfo)
+
+    return int(parsed_date.timestamp())
+
+
+def parse_date(value, tzinfo=datetime.timezone.utc):
+    """Parse an arbitrary date value into a Python datetime.
+
+    If no value is provided (i.e., the value is None or empty), then the return value will be
+    None.
+
+    `Args:`
+        value: int or str or datetime
+            Value to parse
+        tzinfo: datetime.timezone
+            `Optional`: Timezone for the datetime; defaults to UTC.
+    `Returns:`
+        datetime.datetime or None
+    """
+
+    if not value:
+        return None
+
+    # If it's a number, we (probably) have a unix timestamp
+    if isinstance(value, int):
+        parsed = datetime.datetime.fromtimestamp(value, tzinfo)
+    elif isinstance(value, datetime.datetime):
+        parsed = value
+    elif isinstance(value, str):
+        parsed = parse(value)
+    else:
+        raise TypeError(
+            'Unable to parse value; must be one of string or int or datetime, but got type '
+            f'{type(value)}')
+
+    if not parsed.tzinfo:
+        parsed = parsed.replace(tzinfo=tzinfo)
+
+    return parsed
+
+
+def timestamp_to_readable(value, tzinfo=datetime.timezone.utc, format_as='%Y-%m-%d %H:%M:%S'):
```

Comment:
Why not use the ISO format as the default format?

Reply:
i.e. the built-in one?

Reply:
ISO is a little less readable, though certainly more readable than unix timestamps! But it might be worth the tradeoff to use a standardized value.